### ISSUE TITLE
Automated combability analysis for SSP and FMUs in OMEdit

### ISSUE THREAD

**Issue by AnHeuermann** [OpenModelica/OpenModelica#7168]
## Is your feature request related to a problem? Please describe.
When getting a FMU or SSP I have to ask (or guess) what platform the FMUs can run on. From OMEdit the only way (I could find) is to simply simulate the SSP model and look at the error messages to see that the FMU is missing the binaries for my platform.

That's not very convenient for users who don't know much about FMI.

```
[3] 10:12:25 Scripting Error
[fmiLogger] module FMILIB: Could not change to the DLL directory C:/Users\Andreas\AppData\Local\Temp\OpenModelica\OMEdit\TwoCapacitors-0kmfhs1q\temp\0001_Modelica_Blocks_Sources_Sine\binaries\win64\

[4] 10:12:25 Scripting Error
[fmiLogger] module FMILIB: System error: No error

[5] 10:12:25 Scripting Error
[instantiate] Could not load "resources/0001_Modelica_Blocks_Sources_Sine.fmu" which is associated with "TwoCapacitors.Root.Sine"; it may be corrupted or may not support your platform

[6] 10:12:25 Scripting Error
[terminate] Model "TwoCapacitors" is in wrong model state
```

## Describe the solution you'd like
We already have [getFMUInfo](https://openmodelica.org/doc/OpenModelicaUsersGuide/latest/omsimulator.html#getfmuinfo) which returns `oms_fmu_info_t`, see [Types.h#L407-L531](https://github.com/OpenModelica/OMSimulator/blob/master/src/OMSimulatorLib/Types.h#L407-L531). As far as I can tell the information about available platforms are missing there. But we can fix that qucikly.

We need the same for SSP. 

The OMEdit could then automatically call `getSSPInfo` when loading a SSP model and issue a warning, if your platform is not supported by the SSP (or the FMUs inside the SSP).

**Describe alternatives you've considered**
We can just try to open the .dll or .so with FMILIB and don't rely on `getFMUInfo`.

**Comment by AnHeuermann** [comment]
@adeas31 If I add `getSSPinfo` to OMSimulator, would it be much effort to implement this feature in OMEdit?

**Comment by lochel** [comment]
@AnHeuermann What do you want to return with `getSSPInfo`?

**Comment by AnHeuermann** [comment]
Basically SSPs have multiple FMUs + general stuff, so something like

```C
typedef struct {
  oms_fmu_info_t** fmus // Array of pointerto fmu info inside SSP;
  cosnt char** crefs // Names of fmus from above.

  // General stuff
  char* author;
  char* copyright; 
  /* ... */
}oms_ssp_info_t;
```

**Comment by AnHeuermann** [comment]
Also displaying all the information from `getFMUInfo` and `getSSPInfo` in OMEdit would be great. E.g. in the right-click menu in the Libraries Browser where the SSP model is listed. We could add an item `Information` like we have for Modelica models.

**Comment by lochel** [comment]
The fmu info is already available in OMEdit. Just double click on an fmu 😊

**Comment by adeas31** [comment]
I am not sure if `getSSPInfo` is really needed. Even if we add this the actual problem still remains. I mean we already have `getFMUInfo` but you still end up creating this issue. What we need is that OMSimulator should issue a warning when an FMU is added that is not compatible with the user OS.

**Comment by AnHeuermann** [comment]
That's a good idea. Just checking if the binary directory for e.g. Windows is there or trying to load the dll would be enough to see if it can run on your system.

**Comment by adeas31** [comment]
Yes. And we should just issue warning or a notification so that the user knows that.

In case when user loads a SSP with non-compatible FMUs then OMSimulator can also issue a warning/notification just once on loading.
